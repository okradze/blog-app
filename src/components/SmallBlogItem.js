import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const SmallBlogItem = ({ title, id, createdAt }) => (
	<Link className="recent-feed-item" to={`read/${id}`}>
		<h4 className="recent-feed-item__title">{title}</h4>
		<time className="recent-feed-item__date" dateTime={new Date(createdAt)}>
			{moment(createdAt).format('MMM Do, YYYY')}
		</time>
	</Link>
);

export default SmallBlogItem;
