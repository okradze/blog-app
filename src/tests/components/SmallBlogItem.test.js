import React from 'react';
import { shallow } from 'enzyme';
import SmallBlogItem from '../../components/SmallBlogItem';

let title, id, createdAt;

beforeEach(() => {
	title = 'Title';
	id = 'abc123';
	createdAt = 120000;
});

test('should render component correctly', () => {
	const wrapper = shallow(<SmallBlogItem title={title} id={id} createdAt={createdAt} />);
	expect(wrapper).toMatchSnapshot();
});
