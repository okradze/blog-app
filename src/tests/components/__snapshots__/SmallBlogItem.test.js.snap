// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<Link
  className="recent-feed-item"
  replace={false}
  to="read/abc123"
>
  <h4
    className="recent-feed-item__title"
  >
    Title
  </h4>
  <time
    className="recent-feed-item__date"
    dateTime={1970-01-01T00:02:00.000Z}
  >
    Jan 1st, 1970
  </time>
</Link>
`;
